toolchain/lower/file_context.h

```diff
@@ -28,6 +29,47 @@ class FileContext {
     int32_t column_number;
   };
 
+  struct LoweringFunctionState {
+    // The index of an instruction for the whole function
+    // Note: A more accurante representation would be
+    // index_for_block_id + index+inst_id_in_block. This doesn't look necessary
+    // for deduplicating specifics of the same generic, but would be needed for
+    // more complex analysis (ICF).
+    unsigned inst_id_index = 0;
+    // If the state has relevant information, it will be kept, otherwise it's
+    // discarded.
+    bool relevant = false;
```

Comment:
What's the difference between "relevant" and checking whether all the vectors are empty?